packages/ember-metal/lib/property_events.js

```diff
@@ -110,8 +105,7 @@ function propertyDidChange(obj, keyName, _meta) {
     obj[PROPERTY_DID_CHANGE](keyName);
   }
 
-  if (hasMeta) {
-    if (meta.isSourceDestroying()) { return; }
+  if (hasMeta && !meta.isSourceDestroying()) {
```

Comment:
this will no longer eagerly return, does that not change the behaviour of the function?